tests/validation/tests/v3_api/test_workload.py

```diff
@@ -858,3 +858,333 @@ def setup_project_by_role(role, remove_resource):
+        with pytest.raises(ApiError) as e:
+            p2 = rbac_get_unshared_project()
+            ns2 = rbac_get_unshared_ns()
+            new_p_client = get_project_client_for_token(p2, user_token)
+            new_p_client.create_workload(name=name,
+                                        containers=con,
+                                        namespaceId=ns.id,
+                                        daemonSetConfig={})
+        assert e.value.error.status == 403
+        assert e.value.error.code == 'Forbidden'
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", [CLUSTER_MEMBER,
+                                  PROJECT_MEMBER, PROJECT_OWNER,
+                                  PROJECT_READ_ONLY, CLUSTER_OWNER])
+def test_rbac_wl_daemonset_list(remove_resource):
+    if role == CLUSTER_MEMBER:
+        user_token = rbac_get_user_token_by_role(role)
+        project, ns = create_project_and_ns(user_token, namespace["cluster"],
+                                            random_test_name("test-rbac"))
+        p_client = get_project_client_for_token(project, user_token)
+    else:
+        project = rbac_get_project()
+        # deploy daemonset as cluster owner
+        user_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
+        p_client = get_project_client_for_token(project, user_token)
+        ns = rbac_get_namespace()
+    con = [{"name": "test1",
+            "image": TEST_IMAGE}]
+    name = random_test_name("default")
+    workload = p_client.create_workload(name=name,
+                                        containers=con,
+                                        namespaceId=ns.id,
+                                        daemonSetConfig={})
+    wait_for_wl_to_active(cluster_owner_p_client, workload)
+    # switch to rbac role
+    user_token = rbac_get_user_token_by_role(role)
+    p_client = get_project_client_for_token(project, user_token)
+    assert len(p_client.list_workload().data) == 1
+    remove_resource(workload)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", [CLUSTER_MEMBER,
+                                  PROJECT_MEMBER, PROJECT_OWNER,
+                                  PROJECT_READ_ONLY, CLUSTER_OWNER])
+def test_rbac_wl_daemonset_list_negative(remove_resource):
+    # roles can NOT see workloads in the project they have no access to
+    if role != CLUSTER_OWNER OR CLUSTER_MEMBER:
+        p2 = rbac_get_unshared_project()
+        user_token = rbac_get_user_token_by_role(role)
+        p_client = get_project_client_for_token(p2, user_token)
+        workloads = p_client.list_workload().data
+        assert len(workloads) == 0
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", [CLUSTER_MEMBER,
+                                  PROJECT_MEMBER, PROJECT_OWNER,
+                                  PROJECT_READ_ONLY, CLUSTER_OWNER])
+def test_rbac_wl_daemonset_update(remove_resource):
+    if role == CLUSTER_MEMBER:
+        user_token = rbac_get_user_token_by_role(role)
```

Comment:
There are a few instances of this code block that could be wrapped in a function.

Reply:
Look for other candidates too :)